### DISCUSSION TITLE
how to merge two images

### DISCUSSION THREAD

**Discussion by walksanatora** [ImageMagick/ImageMagick#5909] [Help]
### ImageMagick version

7.1.0-55

### Operating system, version and so on

Linux (arch) CLI

### Description

so i want to composite two images
base.png is a base (64x64) pixels
and overlay.png also is 64x64
but overlay.png has pixels which overlap fully hidden pixels
i was able to find [https://github.com/ImageMagick/ImageMagick/discussions/5395](https://github.com/ImageMagick/ImageMagick/discussions/5395) but i dont know how this works if the base does not ave pixels matching with the overlay
(i basically want to keep pixels that are fully transparent fully transparent after the overlay)

**Comment by walksanatora** [comment]
here are the files i have
base.png
![base](https://user-images.githubusercontent.com/47339836/209612948-96ef6e2d-391e-4a43-98bd-1f2a36d814cf.png)
overlay.png
![overlay](https://user-images.githubusercontent.com/47339836/209612968-a113dfde-1a10-4c41-8672-76ab960e79e4.png)
resulting png 
![result1](https://user-images.githubusercontent.com/47339836/209612984-f503e3fc-3bd4-4fb9-9029-8c3561c0beb3.png)
expected result
![overlay](https://user-images.githubusercontent.com/47339836/209613014-0f6b4971-a439-4a05-a429-6c234a8d5886.png)

**Comment by fmw42** [comment]
Post your two images. I am not quite sure I understand what you want to do.  Do both have transparency?  What do you mean by overlap?  What do you mean by pixels do not match?

**Comment by fmw42** [comment]
How did you create or get the expected results image?  What command did you run in Imagemagick?

**Comment by walksanatora** [comment]
the command i ran in imagemagick is
`magick base.png overlay.png -define compose:args=25,100 -compose dissolve -composite base-out.png `
how i got the expected output
i put it in piskel (which doesen't support transparency so it just deletes semi-transparent pixels)

**Reply by fmw42** [reply]
You had to do something further to it other than just removing transparency.

**Reply by walksanatora** [reply]
all i did was take the resulting image and upload it to https://www.piskelapp.com/
and downloaded it again
and it gave me the result i wanted

**Comment by fmw42** [comment]
Try 

`magick base.png overlay.png -define compose:args=25,75 -compose blend -composite result.png`

**Reply by walksanatora** [reply]
nope
this looks practically the same
note: i just tried `convert base-out.png gif:- | convert - base-out2.png` (which removes all transparent pixels)
and it works
(this is a seperate command so it is annoying to have to do)

**Reply by fmw42** [reply]
All you are doings making. the transparency binary rather than 8-bits. So do this

`magick base.png overlay.png -define compose:args=25,100 -compose dissolve -composite -channel a -threshold 50% +channel x.png`

**Reply by fmw42** [reply]
Or better try

`magick base.png overlay.png -define compose:args=25,100 -compose dissolve -composite PNG8:x.png`

**Reply by walksanatora** [reply]
your second option did not work as it did not overlay properly over the black pixels
your first command does work though

**Comment by walksanatora** [comment]
so the command that i got (which works)
`magick input.png overlay.png -define compose:args=25,100 -compose dissolve -composite gif:- | convert - putput.png`
which works
although a bit annoying (since it involves converting to a diffrent format and back, and piping output to another command
